lib/godmin/engine_wrapper.rb

```diff
@@ -2,32 +2,47 @@ module Godmin
   class EngineWrapper
     attr_reader :engine
 
-    def initialize(obj)
-      @engine = find_engine(obj)
+    def initialize(controller)
+      @engine = find_engine(controller)
     end
 
-    def root
-      engine.root
+    def namespace
+      @namespace ||= engine.railtie_namespace
     end
 
-    def namespace
-      engine.railtie_namespace.to_s.underscore
+    def namespaced?
+      @namespaced ||= namespace.present?
+    end
+
+    def namespaced_path
```

Comment:
We also have the same one here: https://github.com/varvet/godmin/blob/master/lib/godmin/generators/base.rb#L22